pkgs/applications/networking/browsers/firefox/packages.nix

```
@@ -62,6 +63,7 @@
 
   firefox-devedition = buildMozillaMach rec {
     pname = "firefox-devedition";
+    binaryName = "firefox-developer-edition";
```

```suggestion
    pname = "firefox-developer-edition";
```

And I think we could probably fallback to pname if no binaryName is set, since now most of them are matching